### PR TITLE
Fix stats leakage

### PR DESCRIPTION
**Pull request by chrmod**
PR aims to prevent adblocker and anti-tracking stats "leaking" between navigations. Sadly the solution is not complete for Ghostery use-case as the same problem is also present in the Ghostery codebase.

### CODE REVIEW

**Comment by chrmod** [comment]
@philipp-classen @smalluban please review commits in isolation and do not squash on merge.

File: modules/hpn-lite/tests/unit/server-public-key-accessor-test.es [hunk 1]
```diff
@@ -90,16 +90,12 @@ export default describeModule('hpn-lite/server-public-key-accessor',
...
       beforeEach(async function () {
-        /* eslint-disable-next-line global-require */
-        global.crypto = global.crypto || {
-          subtle: {
-            importKey(...args) {
-              return MOCKS.importKey(...args);
-            },
-          }
-        };
+        if (!global.crypto) {
```
Comment (philipp-classen) [inline modules/hpn-lite/tests/unit/server-public-key-accessor-test.es:95]: As it is currently written, the tests have the side effects to change
`global.crypto === undefined` (before tests) to `global.crypto !== undefined` (after tests)

File: modules/adblocker/sources/statistics.es [hunk 2]
```diff
@@ -138,30 +141,51 @@ export default class AdbStats {
   }
 
   init() {
+    browser.webNavigation.onBeforeNavigate.addListener(this.onBeforeNavigate);
```
Comment (philipp-classen) [inline modules/adblocker/sources/statistics.es:144]: I have not tested it, but I wonder if you need to bind the `this` reference? `onBeforeNavigate` uses `this`, but when it is called in that way by the callback, `this` will be in a different context.

Comment (chrmod) [reply modules/adblocker/sources/statistics.es:144]: `onBeforeNavigate` is a class property with arrow function so automatically bound to the instance

**Comment by chrmod** [comment]
closing in favor of smaller PRs

**Pull request closed without merging**